hooks/collector.py

```diff
@@ -319,16 +351,22 @@ def _get_item_info(self, path):
         # keep track if a common type was identified for the extension
         common_type_found = False
 
+        # get the dictionary once to avoid unnecessary processing of icon paths
+        # and other things clients may be doing in subclasses
+        common_file_info = self.common_file_info
```

Comment:
this `common_file_info = self.common_file_info`  feels like premature optimization to me - it should be the responsibility of the `common_file_info` property to handle any caching, memoizing etc.

Reply:
done.